indexer/cli/cli.go

```diff
@@ -28,10 +35,39 @@ func runIndexer(ctx *cli.Context) error {
 		log.Crit("Failed to load config", "message", err)
 	}
 
-	// finish me
+	db, err := database.NewDB(getDsn(conf.DB))
```

Comment:
we dont need this PR anymore since the `Indexer` struct in `indexer.go` is encapsulates the running of the indexer process. Refer to develop

Reply:
Likely should pass in the db through dep injection instead of encapsalating the db

Reply:
Ah this comment wasn't related to this line. I was just generally commenting this i realized i approved this PR a long time ago before I had go the indexer processor wired up with the CLI.

Makes sense if you want the database to consume the database config object rather than create it. Since this PR hash a lot of stale history, i'd just close this implement that in a seperate pr

Reply:
Yea same page came back to this pr to close it in favor of https://github.com/ethereum-optimism/optimism/pull/6771 which both does the dep injection for the db but also makes it so anything consuming the config only consumes the part of the config it needs explicitly